scenarioo-server/src/main/java/org/scenarioo/business/builds/BuildImporter.java

```diff
@@ -120,14 +120,20 @@ public synchronized int submitUnprocessedBuildsForImport(AvailableBuildsList ava
 		List<BuildImportSummary> buildsSortedByDateDescending = BuildByDateSorter
 			.sortBuildsByDateDescending(buildImportSummaries.values());
 
-		List<BuildIdentifier> importNeededBuilds = new LinkedList<BuildIdentifier>();
+		List<BuildIdentifier> importNeededBuilds = new LinkedList<>();
 		for (BuildImportSummary buildImportSummary : buildsSortedByDateDescending) {
 			if (buildImportSummary != null && buildImportSummary.getStatus().isImportNeeded()) {
 				importNeededBuilds.add(buildImportSummary.getIdentifier());
 				submitBuildForImport(availableBuilds, buildImportSummary.getIdentifier());
 			}
 		}
 
+		//After all Builds were imported and all Comparisons triggered, we want to log a successful completion message.
+		//Since the scheduling of comparisons only  happens after all builds are imported, we need to add a second
+		//Job into the queue to ensure that it is triggered last.
+		asyncBuildImportExecutor.submit(() ->
+			asyncBuildImportExecutor.submit(() -> comparisonExecutor.scheduleComparisonFinishedLog()));
```

Comment:
I would put this in a separate method, it seems less distractive for the flow of the logic (because of the comment)

Reply:
Also, is this correct that the submit happens twice?

Reply:
1. Done.
2. Yes, because the first submit is triggered after all Builds are imported but before the Comparisons are processed.

The flow looks like this:
 1. All Builds are executed and every Build schedules his Comparisons with the AsyncBuildImporterExecutor to ensure that they are only processed after all Builds have been imported.
 2. The first submit of submitImportFinishedJob is triggered, it schedules another submit, since the Comparisons have not finished yet.
 3. All Comparisons are scheduled in ComparisonExecutor
 4. The second submit of submitImportFinishedJob is triggered, it schedules a submit in ComparisonExecutor.
 5. ComparisonExecutor processes all Comparisons
 6. ComparisonExecutor processes scheduleComparisonFinishedLog.